raster/src/main/scala/geotrellis/raster/summary/polygonal/PolygonalSummaryResult.scala

```diff
@@ -0,0 +1,8 @@
+package geotrellis.raster.summary.polygonal
+
+sealed trait PolygonalSummaryResult[+A]
+
+case object NoIntersection extends PolygonalSummaryResult[Nothing]
+case class Summary[A](value: A) extends PolygonalSummaryResult[A]
```

Comment:
Are there more states that this is going to represent? Just two seem to map so tightly on `Option` that it raises an eye-brow. I know there are some discussions that have been had on `Option` vs ADT so I may be missing future plans here.

Reply:
No, @pomadchin @moradology and I weren't able to theorycraft additional states that this might represent in the future (but it could). The key thing that this provides is that it provides additional clarity into the result types since the NoIntersection case isn't really an error, its a valid result state. It also avoids weird types such as `Option[Option[Double]]` for the `TileFoldingVisitor` class of summaries. Instead you get `PolygonalSummaryResult[Option[Double]]` which we all though provided more clarity.